courses/forms/subscribe_form.py

```diff
@@ -116,6 +116,17 @@ def clean(self) -> dict:
                     return cleaned_data
 
                 partner = User.objects.get(email=partner_email)
```

Comment:
```suggestion
                partner = User.objects.filter(email=partner_email).first()
```

If there are no results that match the query, [get()](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.get) will raise a DoesNotExist exception. 

For this code to, work `first` should be used. Doc `first`: Returns the first object matched by the queryset, or None if there is no matching object.